server/splunk/splunk.go

```diff
@@ -126,9 +127,10 @@ func (s *splunk) LoginUser(mattermostUserID string, server string, id string) er
 		s.currentUser = u
 		isNew = false
 	} else {
+		loginData := strings.Split(id, "/")
 		s.currentUser = store.SplunkUser{
 			Server: server,
-			Token:  id,
+			Token:  loginData[1],
```

Comment:
In the code above this, we check if we have previously stored the user using the same `id`. I'm not sure how this results in _not_ needing to supply the token in later calls (as mentioned in the PR description). I would expect the whole string to have to be equal (`username/token` together)

Also I forget if this is a strict limitation of Splunk, but it's unfortunate that there is a "global" user (`s.currentUser`) rather than having a Splunk user per MM user.

Reply:
> If id doesn't have a slash in it, then this code will crash right?

Actually I tested with the username only and it didn't, but I'll double check tomorrow, I wanted to add some tests or checks.

> In the code above this, we check if we have previously stored the user using the same id. I'm not sure how this results in not needing to supply the token in later calls (as mentioned in the PR description). I would expect the whole string to have to be equal (username/token together)

Yep, I'll double check tomorrow and see if there is something missing here, I tested with `username/token` and only  `username` after that, and in both cases worked fine. I think this case (having slash or not) is handled in a level above (command maybe), but I'll handle it here just in case.

Thank you @mickmister !

Reply:
I think I covered everything [here](https://github.com/mattermost/mattermost-plugin-splunk/pull/101/files#diff-0232fe5be5cc9878cfce21f142b18bb0cf18ae275830fa50408148798fbbf3cc) @mickmister 

Let me know what you think!